src/OpenTelemetry.AutoInstrumentation.NLogTarget/OpenTelemetryTarget.cs

```diff
@@ -0,0 +1,211 @@
+            .SetResourceBuilder(ResourceConfigurator.CreateResourceBuilder(Instrumentation.GeneralSettings.Value.EnabledResourceDetectors));
+
+        loggerProviderBuilder = loggerProviderBuilder.AddOtlpExporter(options =>
+        {
+            var endpoint = Endpoint;
+            if (string.IsNullOrEmpty(endpoint))
+            {
+                endpoint = Environment.GetEnvironmentVariable("OTEL_EXPORTER_OTLP_ENDPOINT");
+            }
+
+            if (!string.IsNullOrEmpty(endpoint))
+            {
+                options.Endpoint = new Uri(endpoint!, UriKind.RelativeOrAbsolute);
+            }
+
+            if (!string.IsNullOrEmpty(Headers))
+            {
+                options.Headers = Headers;
+            }
+
+            options.Protocol = UseHttp ? OpenTelemetry.Exporter.OtlpExportProtocol.HttpProtobuf : OpenTelemetry.Exporter.OtlpExportProtocol.Grpc;
+            options.BatchExportProcessorOptions.ScheduledDelayMilliseconds = ScheduledDelayMilliseconds;
+            options.BatchExportProcessorOptions.MaxQueueSize = MaxQueueSize;
+            options.BatchExportProcessorOptions.MaxExportBatchSize = MaxExportBatchSize;
+        });
+
+        _loggerProvider = loggerProviderBuilder.Build();
+        _getLoggerFactory = CreateGetLoggerDelegate(_loggerProvider);
+    }
+
+    protected override void Write(LogEventInfo logEvent)
+    {
+        if (_loggerProvider is null)
+        {
+            return;
+        }
+
+        if (Sdk.SuppressInstrumentation)
+        {
+            return;
+        }
+
+        var logger = GetOrCreateLogger(logEvent.LoggerName);
+
+        // Build properties from event properties and context
+        var properties = new List<KeyValuePair<string, object?>>();
+        if (IncludeEventProperties && logEvent.HasProperties && logEvent.Properties is not null)
+        {
+            foreach (var kvp in logEvent.Properties)
+            {
+                properties.Add(new KeyValuePair<string, object?>(Convert.ToString(kvp.Key)!, kvp.Value));
+            }
+        }
+
+        // Scope properties can be added via explicit <attribute> entries or NLog's contexts (GDC/MDLC)
+        foreach (var attribute in Attributes)
+        {
+            var value = attribute.Layout?.Render(logEvent);
+            if (!string.IsNullOrEmpty(attribute.Name))
+            {
+                properties.Add(new KeyValuePair<string, object?>(attribute.Name!, value));
+            }
+        }
+
+        var body = IncludeFormattedMessage ? logEvent.FormattedMessage : Convert.ToString(logEvent.Message);
+
+        var severityText = logEvent.Level.Name;
+        var severityNumber = MapLogLevelToSeverity(logEvent.Level);
+
+        var current = Activity.Current;
+
+        // Emit using internal helpers via reflection delegate
+        var renderedMessage = logEvent.FormattedMessage;
+        var args = IncludeEventParameters && logEvent.Parameters is object[] p ? p : null;
```

Comment:
Think you should only `IncludeEventParameters` when `logEvent.HasProperties == false`